src/bindings/python/PyGpuShaderDesc.cpp

```diff
@@ -1,186 +1,337 @@
+enum GpuShaderDescIterator
+{
+    IT_UNIFORM = 0,
+    IT_TEXTURE,
+    IT_TEXTURE_3D
 };
 
-}
+using UniformIterator   = PyIterator<GpuShaderDescRcPtr, IT_UNIFORM>;
+using TextureIterator   = PyIterator<GpuShaderDescRcPtr, IT_TEXTURE>;
+using Texture3DIterator = PyIterator<GpuShaderDescRcPtr, IT_TEXTURE_3D>;
 
-///////////////////////////////////////////////////////////////////////////
-///
-
-PyTypeObject PyOCIO_GpuShaderDescType = {
-    PyVarObject_HEAD_INIT(NULL, 0)              //ob_size
-    OCIO_PYTHON_NAMESPACE(GpuShaderDesc),       //tp_name
-    sizeof(PyOCIO_GpuShaderDesc),               //tp_basicsize
-    0,                                          //tp_itemsize
-    (destructor)PyOCIO_GpuShaderDesc_delete,    //tp_dealloc
-    0,                                          //tp_print
-    0,                                          //tp_getattr
-    0,                                          //tp_setattr
-    0,                                          //tp_compare
-    0,                                          //tp_repr
-    0,                                          //tp_as_number
-    0,                                          //tp_as_sequence
-    0,                                          //tp_as_mapping
-    0,                                          //tp_hash 
-    0,                                          //tp_call
-    0,                                          //tp_str
-    0,                                          //tp_getattro
-    0,                                          //tp_setattro
-    0,                                          //tp_as_buffer
-    Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE,   //tp_flags
-    GPUSHADERDESC__DOC__,                       //tp_doc 
-    0,                                          //tp_traverse 
-    0,                                          //tp_clear 
-    0,                                          //tp_richcompare 
-    0,                                          //tp_weaklistoffset 
-    0,                                          //tp_iter 
-    0,                                          //tp_iternext 
-    PyOCIO_GpuShaderDesc_methods,               //tp_methods 
-    0,                                          //tp_members 
-    0,                                          //tp_getset 
-    0,                                          //tp_base 
-    0,                                          //tp_dict 
-    0,                                          //tp_descr_get 
-    0,                                          //tp_descr_set 
-    0,                                          //tp_dictoffset 
-    (initproc) PyOCIO_GpuShaderDesc_init,       //tp_init 
-    0,                                          //tp_alloc 
-    0,                                          //tp_new 
-    0,                                          //tp_free
-    0,                                          //tp_is_gc
+struct Texture
+{
+    const char * textureName;
```

Comment:
What is the lifespan of these raw pointers?
If the pointer is from the C++ implementation, the code should guaranty it in some way!

Reply:
Good call. I changed these to `std::string` for safety.